File: b_rank_skillcheck_sample/long-table_06.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        int[] a = ReadLine().Trim().Split().Select( int.Parse ).ToArray();
        int[] s = ReadLine().Trim().Split().Select( int.Parse ).ToArray();

        foreach( int i in Range( a[1] - 1, a[0] ) ) {
            int t = n <= i ? i - n : i;

            if( s[t] == 1 ) {
                WriteLine( "No" );
                return;
            } else {
                s[t] = 1;
            }
        }
        WriteLine( "Yes" );
        WriteLine( string.Join( " ", s ) );
        return;
    }
}

File: b_rank_skillcheck_sample/long-table_06.py
# 実行時間 10ms
import sys

n = int(input())
a, b = list(map(int, input().split()))
s = list(map(int, input().split()))

for i in range(b, a + b):
    t = i - n - 1 if n <= i else i - 1

    if s[t] == 0:
        s[t] = 1
    elif s[t] == 1:
        print("No")
        sys.exit()
print("Yes")
print(*s)
